crates/oxc_prettier/src/printer/mod.rs

```diff
@@ -220,9 +228,18 @@ impl<'a> Printer<'a> {
         self.line_suffix.push(Command { indent, mode, doc: Doc::Array(docs) });
     }
 
-    fn handle_if_break(&mut self, doc: Doc<'a>, indent: Indent, mode: Mode) {
-        if mode == Mode::Break {
-            self.cmds.push(Command::new(indent, Mode::Break, doc));
+    fn handle_if_break(&mut self, if_break: IfBreak<'a>, indent: Indent, mode: Mode) {
+        let IfBreak { break_contents, flat_content, group_id } = if_break;
+        let group_mode = group_id.map_or(Some(mode), |id| self.group_mode_map.get(&id).copied());
```

Comment:
```suggestion
        let group_mode = group_id.map_or(mode, |id| self.group_mode_map.get(&id).cloned().unwrap_or(Mode::Flat));
```
i don't think this should return an option - we should always print either the contents of `flat` or `break` regardless of whether the groupid was found in the map.

prehaps we should `dbg_assert!` that it exists?

Reply:
prettier for ref -> https://github.com/prettier/prettier/blob/eebf0e4b5ec8ac24393c56ced4b4819d4c551f31/src/document/printer.js#L269-L271

Reply:
oh my bad, was looking at `fits`